examples/common/utility/utility.hpp

```diff
@@ -356,6 +358,62 @@ class cli_argument_pack {
     }
 }; // class cli_argument_pack
 
+// utility class to aid relative error measurement of samples
+class measurements {
+public:
+    measurements() = default;
+
+    measurements(int iterations) {
+        _time_intervals.reserve(iterations);
+    }
+
+    inline void start() {
+        _startTime = std::chrono::steady_clock::now();
+    }
+    inline void stop() {
+        auto _endTime = std::chrono::steady_clock::now();
+        // store the end time and start time
+        _time_intervals.push_back(std::make_pair(_startTime, _endTime));
+    }
+    double computeRelError() {
+        // Accumulate the total duration in microseconds using std::accumulate with a lambda function
+        if (0 == _time_intervals.size()) {
+            std::cout << "No time samples collected \n";
+            return 0;
+        }
+        auto total_duration = std::accumulate(
+            _time_intervals.begin(),
+            _time_intervals.end(),
+            0, // Start with 0 count
+            [](long long total, const std::pair<time_point, time_point>& interval) {
+                // Compute the difference and add it to the total
+                return total + std::chrono::duration_cast<std::chrono::microseconds>(
+                                   interval.second - interval.first)
+                                   .count();
```

Comment:
I wonder if it would be better to optimize by computing the `std::chrono::duration_cast<std::chrono::microseconds>(interval.second - interval.first).count()` only once and reuse the result below. What do you think?

Reply:
Storing the duration at each time point in a vector and reading it back for accumulation and standard deviation calculations increases memory usage. In contrast, calculating the duration on the fly for these functions is more computationally complex but avoids additional memory overhead. Since this code is not on the critical path, I’m fine with the current approach. However, if you think storing durations in a vector for later access is preferable, I can switch to that.